pkg/autoscaler/autoscaler.go

```diff
@@ -334,21 +360,9 @@ var _ = Describe("[Feature:Machines] Autoscaler should", func() {
 		By(fmt.Sprintf("Creating ClusterAutoscaler configured with maxNodesTotal:%v", maxNodesTotal))
 		clusterAutoscaler := clusterAutoscalerResource(maxNodesTotal)
 		Expect(client.Create(context.TODO(), clusterAutoscaler)).Should(Succeed())
+		cleanupObjects[clusterAutoscaler.GetName()] = runtime.Object(clusterAutoscaler)
```

Comment:
Is this cast to a `runtime.Object` necessary? `clusterAutoscaler` is a pointer so should implement the interface already

There are a number of occurrences of this through this file